rust/perspective-viewer/src/rust/components/column_settings_sidebar/attributes_tab.rs

```diff
@@ -0,0 +1,135 @@
+// ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+// ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+// ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+use wasm_bindgen::JsValue;
+use yew::{function_component, html, Callback, Html, Properties};
+
+use crate::components::expression_editor::ExpressionEditor;
+use crate::components::viewer::ColumnLocator;
+use crate::config::ViewConfigUpdate;
+use crate::model::UpdateAndRender;
+use crate::renderer::Renderer;
+use crate::session::Session;
+use crate::utils::ApiFuture;
+use crate::{derive_model, html_template};
+
+#[derive(PartialEq, Clone, Properties)]
+pub struct AttributesTabProps {
+    pub selected_column: ColumnLocator,
+    pub on_close: Callback<()>,
+    pub session: Session,
+    pub renderer: Renderer,
+}
+derive_model!(Renderer, Session for AttributesTabProps);
+
+#[function_component]
+pub fn AttributesTab(p: &AttributesTabProps) -> Html {
+    let is_validating = yew::use_state_eq(|| false);
+    let on_save = yew::use_callback(
+        |v, p| {
+            match &p.selected_column {
+                ColumnLocator::Expr(None) => save_expr(v, p),
+                ColumnLocator::Expr(Some(alias)) => update_expr(alias, &v, p),
+                _ => {
+                    tracing::error!("Tried to save expression in bad state!")
+                }
+            }
+
+            p.on_close.emit(());
+        },
+        p.clone(),
+    );
+
+    let on_validate = yew::use_callback(
+        |b, validating| {
+            validating.set(b);
+        },
+        is_validating.setter(),
+    );
+
+    let on_delete = yew::use_callback(
+        |(), p| {
+            if let ColumnLocator::Expr(Some(ref s)) = p.selected_column {
+                delete_expr(s, p);
+            }
+
+            p.on_close.emit(());
+        },
+        p.clone(),
+    );
+
+    let alias = yew::use_memo(
```

Comment:
This doesn't need to be a `use_memo`, just calculate it in the component body. In this case it's not even conserving the `clone()` as you must clone into the closure.